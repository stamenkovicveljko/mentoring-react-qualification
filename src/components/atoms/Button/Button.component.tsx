import * as React from 'react';
import { StyledButton } from "./Button.styles";

interface Props {
  type?: "button" | "submit" | "reset";
  onClick?: (ev: React.ChangeEvent<HTMLButtonElement>) => void;
  variant?: "standard" | "primary" | "secondary";
  children?: React.ReactNode;
}

function Button({
  type = "button",
  variant = "standard",
  onClick = () => {},
  children,
}: Props): React.ReactElement {
  return (
    <StyledButton<any> onClick={onClick} variant={variant} type={type}>
      {children}
    </StyledButton>
  );
}

export default Button;
