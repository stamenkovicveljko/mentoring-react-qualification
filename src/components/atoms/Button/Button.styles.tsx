import styled, { css } from "styled-components";
import { resetButton } from "../../../styles/resets";
import { fontSize } from "../../../styles";

export const StyledButton = styled.button`
  ${resetButton}
  ${fontSize("16px", "20px")}
  cursor: pointer;
  padding: 10px;
  border: solid 1px #212121;
  border-radius: 4px;

  ${({ variant }: any) => {
    const standard = css`
      color: #212121;
      background-color: white;
      border-radius: 4px;
      background-color: #fafafa;
      &:active {
        background-color: #eaeaea;
      }
    `;

    switch (variant) {
      case "primary":
        return css`
          background-color: #0366d6;
          border-color: #0366d6;
          color: white;
        `;

      case "secondary":
        return css`
          padding: 0;
          border: none;
          color: #0366d6;
          &:hover {
            text-decoration: underline;
          }
        `;

      default:
        return standard;
    }
  }}
`;
