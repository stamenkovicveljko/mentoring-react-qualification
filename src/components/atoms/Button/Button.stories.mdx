import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import Button from "./Button.component";

<Meta title="Button" component={Button} />

# Button Component

This is a pretty straight forward button that comes with the variants sufficient for the project.

## Implementation

<Preview>
  <Story name="default">
    <Button>Default(standard) button</Button>
  </Story>
</Preview>

<Preview>
  <Story name="primary">
    <Button variant="primary">Primary button</Button>
  </Story>
</Preview>

<Preview>
  <Story name="secondary">
    <Button variant="secondary">Secondary button</Button>
  </Story>
</Preview>

## Props

<Props of={Button} />
