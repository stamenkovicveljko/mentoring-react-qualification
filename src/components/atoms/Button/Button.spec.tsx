import React from "react";
import renderer from "react-test-renderer";
import Button from "./Button.component";

test("Button renders as expected", () => {
  const component = renderer.create(
    <Button variant="primary">Sample button</Button>
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
