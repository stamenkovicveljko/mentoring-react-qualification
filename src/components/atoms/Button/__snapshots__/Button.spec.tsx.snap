// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders as expected 1`] = `
<button
  className="sc-AxjAm vJGYS"
  onClick={[Function]}
  type="button"
>
  Sample button
</button>
`;
